server/server.go

```diff
@@ -133,21 +137,37 @@ func (h *SlackHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			// It's a command but we have no handler for it - 404
 			serve(h.DefaultRoute, nil)
 			return
-		}
-		// Attempt to parse as an event
-		event, err := h.actionEventHelper(r)
-		if err != nil {
-			h.Log("Error parsing Slack Event: %s", err)
-		}
-		if len(event.CallbackId) > 0 {
+		} else {
+			var payloadMap map[string]interface{}
+			payloadJSON := r.Form.Get("payload")
+			if err := json.Unmarshal([]byte(payloadJSON), &payloadMap); err != nil {
+				h.Log("%s", err)
+				serve(h.DefaultRoute, nil)
+				return
+			}
+			if payloadMap["type"] == nil {
+				h.Log("Error parsing Slack Event: %s", "Missing value for 'type' key")
+				serve(h.DefaultRoute, nil)
+				return
+			}
+			if payloadMap["callback_id"] == nil {
+				h.Log("Error parsing Slack Event: %s", "Missing value for 'callback_id' key")
```

Comment:
Pointless interpolation:
 
Try 

`h.Log("Error parsing Slack Event: Missing value for 'callback_id' key")`